### PR TITLE
feat: withdraw

### PR DESCRIPTION
**Pull request by anvztor**

### CODE REVIEW

File: internal/state/eventbus.go [hunk 1]
```diff
@@ -18,11 +18,12 @@ const (
 	SigFailed
 	SigTimeout
 	DepositReceive
+	WithdrawRequest
```
Comment (aiden069) [inline internal/state/eventbus.go:21]: Please add  it after BlockScanned, enum index cannot be replace

File: internal/utxo/withdraw.go
Comment (aiden069) [inline internal/utxo/withdraw.go:1]: Move this file to wallet folder, delete utxo folder if it is possible

File: internal/state/eventbus.go (hunk 1 above)
Comment (anvztor) [reply internal/state/eventbus.go]: fixed.

File: internal/layer2/consensus_event.go [hunk 2]
```diff
@@ -35,9 +35,15 @@ func (lis *Layer2Listener) processEvent(block uint64, event abcitypes.Event) err
 		return lis.processNewWalletKey(block, event.Attributes)
 	case bitcointypes.EventTypeNewDeposit:
 		return lis.processNewDeposit(block, event.Attributes)
+	case bitcointypes.EventTypeWithdrawalCancellation:
```
Comment (aiden069) [inline internal/layer2/consensus_event.go:38]: rename to processUser***

File: internal/btc/btcutils.go [hunk 3]
```diff
@@ -86,3 +91,90 @@ func VerifyTransaction(tx wire.MsgTx, txHash string, evmAddress string) error {
 
 	return nil
 }
+
+func P2wshDeposit(netwk *chaincfg.Params, tssGroupKey *btcec.PrivateKey, evmAddresses [][]byte, prevTxIds []string,
```
Comment (aiden069) [inline internal/btc/btcutils.go:95]: rename to SpendP2wsh
rename and move file to wallet

**Pull request merged**